packages/control-plane/src/__tests__/credential-service.test.ts

```
@@ -0,0 +1,468 @@
+      const returningAll = vi.fn().mockReturnValue({ executeTakeFirstOrThrow })
+      const whereFn: ReturnType<typeof vi.fn> = vi.fn()
+      const whereChain = { where: whereFn, execute, returningAll }
+      whereFn.mockReturnValue(whereChain)
+      const set = vi.fn().mockReturnValue(whereChain)
+      return { set }
+    }),
+
+    deleteFrom: vi.fn().mockImplementation(() => {
+      const execute = vi.fn().mockResolvedValue(undefined)
+      const whereFn: ReturnType<typeof vi.fn> = vi.fn().mockReturnValue({
+        where: whereFn,
```

Biome (2.4.4)

[error] This variable is used before its declaration.

(lint/correctness/noInvalidUseBeforeDeclaration)